### PR TITLE
Add documentation to WorkForm component

### PR DESCRIPTION
**Pull request by rioc0719**
### Problem
File src/client/components/forms/work.js had minimal documentation
beforehand. 

### Solution
Added comments to clarify the purpose of the methods and
some bits of code that were ambiguous at first.

### Areas of Impact
src/client/components/forms/work.js was changed. No behavior is affected.

I wasn't able to find the precommit hook but I attempted to remain consistent
with the coding style as I could.

(For GCI 2016-17)

### CODE REVIEW

File: src/client/components/forms/work.js [hunk 1]
```diff
@@ -112,6 +140,12 @@ class WorkForm extends React.Component {
 			});
 	}
 
+	/**
+	 * Renders the form component to allow the user to input
+	 * and submit data about a Work. Also shows a loading spinner
+	 * if connectivity causes a delay in submission.
+	 * @return {ReactElement} - react component (HTML) for a workeditor
```
Comment (LordSputnik) [inline src/client/components/forms/work.js]: I'd change this to "React element corresponding to the rendered WorkEditor component"

Comment (rioc0719) [inline src/client/components/forms/work.js]: Alright, I made the change!

**Comment by LordSputnik** [comment]
Almost there, but run 'npm run lint' and fix the 1 error from this PR.

**Comment by rioc0719** [comment]
I'm getting different errors from `npm run lint` than from the Travis build, but no errors in the file I modified. Is that normal?

**Comment by LordSputnik** [comment]
Hmm, it's possibly down to different versions of eslint - if you run `eslint -v` or `./node_modules/.bin/eslint -v`, what do you get?

**Comment by LordSputnik** [comment]
FYI, your docs are now public at https://doclets.io/bookbrainz/bookbrainz-site/master - that'll update automatically whenever any new ones are added!

**Comment by rioc0719** [comment]
My global version of Eslint is 3.13.1, but I never installed a local version for some reason even though it's in the package.json. And that's great, I'll try to fill it up with some docs soon!

**Pull request merged**